### PR TITLE
Fix UseCompilerGeneratedDocXmlFile default value

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes what was reported in https://github.com/dotnet/runtime/issues/105974#issuecomment-2274090272 by @michaelgsharp (thanks!)

Regressed with https://github.com/dotnet/runtime/pull/94234

For now suppress the undocumented public APIs that got added since the property got switched to the wrong default. These public APIs are tracked via https://github.com/dotnet/runtime/issues/105974.

@carlossanlop @ericstj please take a look and merge

### CODE REVIEW

**Review by carlossanlop — COMMENTED** [review]
For Tensors, instead of pragma disables, can we instead disable the property in the csproj? The APIs are actively getting documented, so we should keep it simple and easy to activate, as well as try to avoid a bunch of merge conflicts https://github.com/dotnet/runtime/pull/106084

**Comment by ViktorHofer** [comment]
Disabling the property in the project file is probably not what we want. That stops the compiler from emitting CS1591 for new API that is getting added or from validating that existing documented API stays documented.

File: eng/intellisense.targets [hunk 1]
```diff
@@ -1,5 +1,9 @@
 <Project>
 
+  <PropertyGroup>
+    <UseCompilerGeneratedDocXmlFile Condition="'$(UseCompilerGeneratedDocXmlFile)' == ''">true</UseCompilerGeneratedDocXmlFile>
+  </PropertyGroup>
+
   <PropertyGroup Condition="'$(UseCompilerGeneratedDocXmlFile)' != 'true'">
```
Comment (ericstj) [inline eng/intellisense.targets:7]: We should add an error below in `ChangeDocumentationFileForPackaging` for when `UseCompilerGeneratedDocXmlFile` is not `true` and `IntellisensePackageXmlFilePath` is not set (the file doesn't exist).  That will prevent the case where folks think they're getting one thing but aren't.

Comment (ericstj) [reply eng/intellisense.targets:7]: In other words, remove `'$(IntellisensePackageXmlFilePath)' != ''` from the target condition, add it to the ItemGroup, then add a warning that's the inverse:

```
<Warning Condition="'$(IntellisensePackageXmlFilePath)' == ''" Text="`UseCompilerGeneratedDocXmlFile` was set to `$(UseCompilerGeneratedDocXmlFile)`, but the doc-team provided file was not found" />
```

Comment (ViktorHofer) [reply eng/intellisense.targets:7]: Done. I added the warning to a separate target though.

File: src/libraries/System.Speech/src/System.Speech.csproj [hunk 2]
```diff
@@ -7,8 +7,9 @@
     <!-- CS0649: uninitialized interop type fields -->
     <!-- SA1129: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3277 -->
     <NoWarn>$(NoWarn);CS0649;SA1129;IDE0059;IDE0060;CA1822;CA1852</NoWarn>
+    <!-- TODO: Document public visible types and members. https://github.com/dotnet/runtime/issues/87711 -->
+    <NoWarn>$(NoWarn);CS1591</NoWarn>
     <IsTrimmable>false</IsTrimmable>
-    <UseCompilerGeneratedDocXmlFile>false</UseCompilerGeneratedDocXmlFile>
```
Comment (ericstj) [inline src/libraries/System.Speech/src/System.Speech.csproj:11]: Looks like this is causing an error due to it's use of a PNSE.

**Review by ericstj — APPROVED** [review]
LGTM, once we get System.Speech working correctly.  Might have to manually override the check on this one to allow for no XML for the PNSE (better than no docs at all)

**Pull request merged**